src/include/btree.i

```diff
@@ -1021,12 +1095,10 @@ __wt_page_release(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
 	 * it contains an update that isn't stable.  Also skip forced eviction
 	 * if we just did an in-memory split.
 	 */
-	if (LF_ISSET(WT_READ_NO_EVICT) ||
-	    page->read_gen != WT_READGEN_OLDEST ||
+	if (page->read_gen != WT_READGEN_OLDEST ||
```

Comment:
@michaelcahill, maybe we should just rename `WT_READGEN_OLDEST` to be something like `WT_READGEN_EVICT`?